chimneyCats/src/main/scala/io/scalaland/chimney/cats/TransformationError.scala

```diff
@@ -0,0 +1,31 @@
+package io.scalaland.chimney.cats
+
+import cats.data.{Chain, NonEmptyChain}
+
+abstract class TransformationError(val message: String) {
```

Comment:
1. Is it necessary that this type is abstract? It seems like it could be a case class containing a message and an error path (possibly empty). WDYT?

2. Is it possible to abstract over concrete error message type? Currently in lifted transformers we may just use strings, as good as some custom type that describe error.

Reply:
Made message type abstract, made case class instead of adt